src/fp.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2021 The Authors
+// SPDX-License-Identifier: MPL-2.0
+
+use rand::prelude::*;
+use rand::Rng;
+
+// Parameters of a finite field with prime p < 2^126. Field elements are represented as u128's.
```

Comment:
Rust doccomments should start with `///` so that various documentation tools (like docs.rs) can parse them: https://doc.rust-lang.org/beta/rust-by-example/meta/doc.html

Similarly, the comments next to the fields on `Fp` can be converted into doccomments

Reply:
Done. Do let me know if there's anything else I can do to improve documentation.

Reply:
I changed `mod fp` to `pub mod fp` in `lib.rs` in order to make example code in the docucomments compile. I'm not sure we should actually expose this module in the public API.

Reply:
Reverted this change. `prio::fp` is now private.